studio/src/aem/fragment.js

```diff
@@ -89,6 +89,17 @@ export class Fragment {
         return this.fields.find((field) => field.name === 'originalId');
     }
 
+    isVariation() {
+        const originalIdField = this.getOriginalIdField();
+        const originalId = originalIdField?.values?.[0];
+        return !!originalId && originalId !== this.id;
+    }
+
+    getParentId() {
+        const originalIdField = this.getOriginalIdField();
+        return originalIdField?.values?.[0] || null;
+    }
+
```

Comment:
This will now work unfortunately. originalId has been added to differentiate a US fragment from a ROW fragment. Here we need to deduce parent path & id from the actual variant path (if locale is region, take default country's)